feat: log retention policies chapter (#606)
italia/eidas-it-wallet-docs: docs/en/log-retention-policy.rst

For all about log handling, Wallet Providers, Credential Issuers, and Relying Parties are considered Organizational Entities. 

Logs related to Wallet data exchange activities (accesses, transactions, credential issuance/revocation) concerning the End User, as the data subject, MUST be retained for a limited period for security, fraud prevention, dispute resolution, and legal obligations.

Organizational Entities are responsible for log retention according to their respective roles. Solutions related to Wallet Providers, Credential Issuers, and Relying Parties MUST implement audit logging for the activities of administrators and service operators with access to data exchange processes and logs.

Unless specific legal obligations dictate otherwise, and with the definition of sector-specific regulation defining appropriate motivations, **the maximum retention period for data logs is 12 months**. Logs MUST be securely stored to ensure integrity and immutability.

> 
> 
> 
> 

Author: ```suggestion
```

> 
> 
> 
> 
> 

Author: ```suggestion
```

For all about log handling, Wallet Providers, Credential Issuers, and Relying Parties are considered Organizational Entities. 

Logs related to Wallet data exchange activities (accesses, transactions, credential issuance/revocation) concerning the End User, as the data subject, MUST be retained for a limited period for security, fraud prevention, dispute resolution, and legal obligations.

Organizational Entities are responsible for log retention according to their respective roles. Solutions related to Wallet Providers, Credential Issuers, and Relying Parties MUST implement audit logging for the activities of administrators and service operators with access to data exchange processes and logs.

Unless specific legal obligations dictate otherwise, and with the definition of sector-specific regulation defining appropriate motivations, **the maximum retention period for data logs is 12 months**. Logs MUST be securely stored to ensure integrity and immutability.

GDPR Logs General Requirements

> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Author: ```suggestion
```

- Credential Issuers MUST log the Issuer signed part of the issued Credentials, such as the Issuer-Signed-JWT using SD-JWT-VC or the MSO using MDoc CBOR.
- Credential Issuer SHOULD NOT log the User personal attributes, such as the Credential disclosure map. 
- Credential Issuers MUST log the Holder's signed revocation request.  

Relying Party Log Retention Policy
----------------------------------

Relying Parties MAY retain logs related to the processing of data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.

At the end of the retention period, data MUST be deleted or anonymized.

In detail, Relying Party are required to be compliant with the following items:

- Relying Parties MUST log Wallet's signed presentations, linked to signed presentation requests, containing the Credential Signed part and the Wallet Signed part, such as the Issuer-Signed-JWT and the KB-JWT using SD-JWT-VC.

> - Relying Parties SHOULD not log Credetnail presentation disclosure map, where not necessary. 

Author: ```suggestion
- Relying Parties SHOULD not log Credential presentation disclosure maps, where not necessary. 
```

.. include:: ../common/common_definitions.rst

.. _log-retention-policy.rst:

> General Log and Data Retention Policies

Reviewer: ```suggestion
General Log Retention Policies
```

=======================================

> The retention of logs and personal data must comply with the principles of the GDPR, the requirements of the eIDAS regulation and the requirements defined in ISO/IEC 27001, with particular attention to data minimization and storage limitation.

Reviewer: ```suggestion
The retention of logs is a key element for ensuring security, including fraud prevention, incident detection, system integrity, and compliance with applicable legal obligations. It MUST also be align with the requirements defined in ISO/IEC 27001, in particular wih regard auditability, access control, and secure storage. As long log management may involve the processing of personal data, it also constitutes a measure of accountability under the GDPR, with implications for data minimization, storage limitation, and purpose limitation; for these aspects, reference is made to the relevant provisions of the GDPR and sector-specific regulations.
```

For all about log handling, Wallet Providers, Credential Issuers, and Relying Parties are considered Organizational Entities. 

Logs related to Wallet data exchange activities (accesses, transactions, credential issuance/revocation) concerning the End User, as the data subject, MUST be retained for a limited period for security, fraud prevention, dispute resolution, and legal obligations.

> Organizational Entities are responsible for data retention according to their respective roles. Solutions related to Wallet Providers, Credential Issuers, and Relying Parties MUST implement audit logging for the activities of administrators and service operators with access to data exchange and logs.

Reviewer: ```suggestion
Organizational Entities are responsible for log retention according to their respective roles. Solutions related to Wallet Providers, Credential Issuers, and Relying Parties MUST implement audit logging for the activities of administrators and service operators with access to data exchange processes and logs.
```

Unless specific legal obligations dictate otherwise, and with the definition of sector-specific regulation defining appropriate motivations, **the maximum retention period for data logs is 12 months**. Logs MUST be securely stored to ensure integrity and immutability.

> GDPR Logs General Requirements

Reviewer: ```suggestion
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> In accordance with the following general principles and in full compliance with them, the retention of logs and personal data made by Organizational Entities MUST adhere to the principles outlined in the GDPR (Article 5), specifically:

Reviewer: ```suggestion
```

> - **Lawfulness, Fairness, and Transparency** (Article 5, Paragraph 1, Letter a): Personal data MUST be processed lawfully, fairly, and in a transparent manner in relation to the data subject.

Reviewer: ```suggestion
```

> - **Purpose Limitation** (Article 5, Paragraph 1, Letter b): Personal data MUST be collected for specified, explicit, and legitimate purposes and not further processed in a manner that is incompatible with those purposes.

Reviewer: ```suggestion
```

> - **Data Minimization** (Article 5, Paragraph 1, Letter c): Personal data MUST be adequate, relevant, and limited to what is necessary in relation to the purposes for which they are processed.

Reviewer: ```suggestion
```

> - **Storage Limitation** (Article 5, Paragraph 1, Letter e): Personal data MUST be kept in a form which permits identification of data subjects for no longer than is necessary for the purposes for which the personal data are processed.

Reviewer: ```suggestion
```

> - **Integrity and Confidentiality** (Article 5, Paragraph 1, Letter f): Personal data MUST be processed in a manner that ensures appropriate security, including protection against unauthorized or unlawful processing and against accidental loss, destruction, or damage, using appropriate technical or organizational measures.

Reviewer: ```suggestion
```

> Organizational Entities MUST implement appropriate technical and organizational measures to ensure the security of logs and personal data, in compliance with Article 32 of the GDPR.

Reviewer: ```suggestion
```

- **A.12.4.2 Protection of Log Information**: Organizational Entities MUST be protected against unauthorized access and tampering. This includes implementing access controls, encryption, and integrity checks to ensure that log data remains secure and reliable.

- **A.12.4.3 Administrator and Operator Logs**: Activities of system administrators and operators MUST be logged and reviewed regularly. Organizational Entities MUST detect unauthorized activities and ensures accountability for actions taken on critical systems.

- **A.12.4.4 Clock Synchronization**: All Organizational Entities' systems involved in logging MUST be synchronized to a reliable time source, ensuring accurate timestamps for correlating events across different systems and conducting effective forensic investigations.

- **A.16.1.7 Information Security Incident Management**: Organizational Entities MUST ensure that logs are available and reliable to support incident management processes.

Wallet Provider Log Retention Policy
------------------------------------

> Personal data related to the registration and management of Wallet Instances can be retained for up to 12 months after the deactivation/revocation of the Wallet or the associated User account, unless legal obligations require longer retention.

Reviewer: ```suggestion
Information related to the registration and management of Wallet Instances can be retained for up to 12 months after the deactivation/revocation of the Wallet or the associated User account, unless legal obligations require longer retention.
```

> Wallet Provider MUST log the following items:

Reviewer: ```suggestion
```

> - Evidences about the User Authentication to the Wallet Provider backends, according to the national regulation pertaining the digital identity systems, if involved (e.g: signed SAML2 Responses, OIDC ID Token). 

Reviewer: ```suggestion
```

> - Wallet Attestation Requests.

Reviewer: ```suggestion
```

> - Wallet Attestations.

Reviewer: ```suggestion
```

> - Holder's requests about the revocation or deactivation of the Wallet Instance.

Reviewer: ```suggestion
```

Credential Issuer Log Retention Policy
--------------------------------------

Credential Issuers define the retention period for Credentials based on sector-specific regulations. In the absence of specific regulations, the retention period for Credentials SHOULD not exceed 12 months after the date of expiration, configured at time of issuance within the metadata of the Issuer Signed part of the Credential.

> In detail, Credential Issuers require to be compliant to the items above:

Reviewer: ```suggestion
```

> - Credential Issuers MUST log the signed requests made by Wallet Instance during the Credential issuance phase, such as authorization requests and token request (Authorization Server) and the Credential requests at the Credential endpoint (Resource Server).

Reviewer: ```suggestion
```

> - Credential Issuers MUST log the evidences about the User Authentication to the Wallet Provider backends, during the authroization phase, according to the national regulation pertaining the digital identity systems, if involved (e.g: signed SAML2 Responses, OIDC ID Token). 

Reviewer: ```suggestion
```

> - Credential Issuers MUST log the Issuer signed part of the issued Credentials, such as the Issuer-Signed-JWT using SD-JWT-VC or the MSO using MDoc CBOR.

Reviewer: ```suggestion
```

> - Credential Issuer SHOULD NOT log the User personal attributes, such as the Credential disclosure map. 

Reviewer: ```suggestion
```

> - Credential Issuers MUST log the Holder's signed revocation request.  

Reviewer: ```suggestion
```

Relying Party Log Retention Policy
----------------------------------

> Relying Parties MAY retain data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.

Reviewer: ```suggestion
Relying Parties MAY retain logs related to the processing of data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.
```

> At the end of the retention period, data MUST be deleted or anonymized.

Reviewer: ```suggestion
```

> In detail, Relying Party are required to be compliant with the following items:

Reviewer: ```suggestion
```

> - Relying Parties MUST log Wallet's signed presentations, linked to signed presentation requests, containing the Credential Signed part and the Wallet Signed part, such as the Issuer-Signed-JWT and the KB-JWT using SD-JWT-VC.

Reviewer: ```suggestion
```

> - Relying Parties SHOULD not log Credetnail presentation disclosure map, where not necessary. 

Reviewer: ```suggestion
```

Wallet Instances Logging Features
---------------------------------

> The Commission Implementing Regulation (EU) 2024/2979 establishes detailed rules for the application of Regulation (EU) No 910/2014, inclusing logging in the core functionalities of European Digital Identity Wallets.

Reviewer: ```suggestion
The Commission Implementing Regulation (EU) 2024/2979 establishes detailed rules for the application of Regulation (EU) No 910/2014, inclusing logging in the core functionalities of European Digital Identity Wallets. Please refer in particular to Recitals (11) and (13), and to Article 5e(4) and Article 6(2)(d), as well as Article 8(2)(d), which define logging obligations, user interface requirements, and data portability mechanisms.
```

> Recital (11) of the regulation highlights the critical role of transaction logging as a tool for transparency. Logs enable Wallet Users to effectively monitor their digital identity activities. This transparency aims to improve the user confidence and accountability in digital identity management.

Reviewer: ```suggestion
```

> In addition to transparency, logs facilitates the swift and use of information with competent supervisory authorities, as outlined in Article 51 of Regulation (EU) 2016/679 (GDPR). In instances of suspicious behavior by Relying Parties, logs can be quickly provided to authorities, enabling prompt investigation and resolution of potential security threats.

Reviewer: ```suggestion
```

> About data export and portability, recital (13) emphasizes the use of data export and portability objects to log person identification data and electronic attestations of attributes issued to a Wallet Instance, supporting the right to data portability. Users are enabled to extract and transfer their data between different Wallet Solutions or recover lost Wallet Instances.

Reviewer: ```suggestion
```

> About data export and portability, recital (13) emphasizes the use of data export and portability objects to log person identification data and electronic attestations of attributes issued to a Wallet Instance, supporting the right to data portability. Users are enabled to extract and transfer their data between different Wallet Solutions or recover lost Wallet Instances.

Reviewer: ```suggestion.
```

> Article 5e, Item 4 of the regulation defines the key features and requirements of the User interface about the navigation of the transaction logs. European Digital Identity Wallets MUST provide a user-friendly, transparent, and traceable dashboard that allows Users to:

Reviewer: ```suggestion
```

> - View an up-to-date list of Relying Parties with which they have established a connection, including all data exchanged where applicable.

Reviewer: ```suggestion
```

> - Easily request the erasure of personal data by a Relying Party, in accordance with Article 17 of Regulation (EU) 2016/679 (GDPR). Users MUST be able to exercise their right to be forgotten, enhancing their control over personal data.

Reviewer: ```suggestion
```

> - Easily report a Relying Party to the competent national data protection authority if an allegedly unlawful or suspicious data request is received, addressing potential data protection violations.

Reviewer: ```suggestion
```

> In detail, Wallet Instances MUST log the following items:

Reviewer: ```suggestion
```

> - Signed presentation requests issued by Relying Parties.

Reviewer: ```suggestion
```

> - Credentials, and disclosed attributes, presented to each Relying Party.

Reviewer: ```suggestion
```

> Regulatory References for Log Retention

Reviewer: ```suggestion
```

---------------------------------------

> - Regulation (EU) 2016/679 (GDPR)

Reviewer: ```suggestion
```

> - Legislative Decree 196/2003 (Personal Data Protection Code)

Reviewer: ```suggestion
```

- ISO/IEC 27001:2024
- Regulation (EU) No 910/2014 (eIDAS)
- National regulations on traffic data retention.

> - Provisions of the Data Protection Authority.

Reviewer: ```suggestion
```

.. include:: ../common/common_definitions.rst

.. _log-retention-policy.rst:

General Log and Data Retention Policies
=======================================

The retention of logs and personal data must comply with the principles of the GDPR, with particular attention to data minimization and storage limitation.
Wallet Providers, Credential Issuers, and Relying Parties must implement appropriate security measures.

Logs related to Wallet data exchange activities (accesses, transactions, credential issuance/revocation) concerning the End User, as the data subject, MUST be retained for a limited period for security, fraud prevention, dispute resolution, and legal obligations.

> In accordance with the following general principles and in full compliance with them, the retention of logs and personal data in digital identity wallets must adhere to the principles outlined in the GDPR (Article 5), specifically:

Reviewer 2: ```suggestion
In accordance with the following general principles and in full compliance with them, the retention of logs and personal data in digital identity wallets MUST adhere to the principles outlined in the GDPR (Article 5), specifically:
```

> In accordance with the following general principles and in full compliance with them, the retention of logs and personal data in digital identity wallets MUST adhere to the principles outlined in the GDPR (Article 5), specifically:

Author: ```suggestion
In accordance with the following general principles and in full compliance with them, the retention of logs and personal data in digital identity Wallets MUST adhere to the principles outlined in the GDPR (Article 5), specifically:
```

- **Purpose Limitation** (Article 5, Paragraph 1, Letter b): Personal data MUST be collected for specified, explicit, and legitimate purposes and not further processed in a manner that is incompatible with those purposes.
- **Data Minimization** (Article 5, Paragraph 1, Letter c): Personal data MUST be adequate, relevant, and limited to what is necessary in relation to the purposes for which they are processed.
- **Storage Limitation** (Article 5, Paragraph 1, Letter e): Personal data MUST be kept in a form which permits identification of data subjects for no longer than is necessary for the purposes for which the personal data are processed.
- **Integrity and Confidentiality** (Article 5, Paragraph 1, Letter f): Personal data MUST be processed in a manner that ensures appropriate security, including protection against unauthorized or unlawful processing and against accidental loss, destruction, or damage, using appropriate technical or organizational measures.

Wallet Providers, Credential Issuers, and Relying Parties MUST implement appropriate technical and organizational measures to ensure the security of logs and personal data, in compliance with Article 32 of the GDPR.

> Unless specific legal obligations dictate otherwise, and with the definition of sector-specific regulation defining appropriate motivations, the maximum retention period for data logs is 24 months. Logs MUST be securely stored to ensure integrity and immutability.

Author: ```suggestion
Unless specific legal obligations dictate otherwise, and with the definition of sector-specific regulation defining appropriate motivations, the maximum retention period for data logs is 12 months. Logs MUST be securely stored to ensure integrity and immutability.
```

Wallet Provider Log Retention Policy
------------------------------------

> Personal data related to the registration and management of Wallet Instances can be retained for up to 24 months after the deactivation/revocation of the Wallet or the associated User account, unless legal obligations require longer retention.

Author: ```suggestion
Personal data related to the registration and management of Wallet Instances can be retained for up to 12 months after the deactivation/revocation of the Wallet or the associated User account, unless legal obligations require longer retention.
```

Credential Issuer Log Retention Policy
--------------------------------------

> Credential Issuers define the retention periods for Credentials based on sector-specific regulations. In the absence of specific regulations, the retention period for credentials should not exceed 5 years from the date of issuance.

Reviewer 2: This period should take into account the time validity period of a Digital Credential. Suppose that a Credential has a validity period greater than 5 years (I know, currently this is not our case) this retention policy could raise some problems.

Author: I need to include ISO 27001 and further details about what needs to be logged (signed tokens, not necessarly with their disclosures) and so on ...

Author: I would remove all these thresholds making a normative base set to 12 months only and give refs to any sectorial-specific regulations

Relying Party Log Retention Policy
----------------------------------

Relying Parties MAY retain data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.

At the end of the retention period, data MUST be deleted or anonymized.

General Responsibilities for Log Retention
------------------------------------------

Wallet Providers, Credential Issuers, and Relying Parties are responsible for data retention according to their respective roles. Solutions related to Wallet Providers, Credential Issuers, and Relying Parties MUST implement audit logging for the activities of administrators and service operators with access to data exchange and logs.

> Regulatory References for Log Retention

Reviewer 2: It is better to include the references in the proper section

Author: yeah, I would got definitively for that once the PR about the "revisited" standards will be completed

Author: this is now definitively out of scope for the releasse v1.0.0

.. include:: ../common/common_definitions.rst

.. _log-retention-policy.rst:

General Log Retention Policies
=======================================

> The retention of logs is a key element for ensuring security, including fraud prevention, incident detection, system integrity, and compliance with applicable legal obligations. It MUST also be align with the requirements defined in ISO/IEC 27001, in particular wih regard auditability, access control, and secure storage. As long log management may involve the processing of personal data, it also constitutes a measure of accountability under the GDPR, with implications for data minimization, storage limitation, and purpose limitation; for these aspects, reference is made to the relevant provisions of the GDPR and sector-specific regulations.

Author: ```suggestion
The retention of logs is a key element for ensuring security, including fraud prevention, incident detection, system integrity, and compliance with applicable legal obligations. It MUST also be align with the requirements defined in ISO/IEC 27001, in particular with regard auditability, access control, and secure storage. As long log management may involve the processing of personal data, it also constitutes a measure of accountability under the GDPR, with implications for data minimization, storage limitation, and purpose limitation; for these aspects, reference is made to the relevant provisions of the GDPR and sector-specific regulations.
```

For all about log handling, Wallet Providers, Credential Issuers, and Relying Parties are considered Organizational Entities. 

> Logs related to Wallet data exchange activities (accesses, transactions, credential issuance/revocation) concerning the End User, as the data subject, MUST be retained for a limited period for security, fraud prevention, dispute resolution, and legal obligations.

Author: ```suggestion
Logs related to Wallet data exchange activities (accesses, transactions, Credential issuance/revocation) concerning the User, as the data subject, MUST be retained for a limited period for security, fraud prevention, dispute resolution, and legal obligations.
```

- **A.16.1.7 Information Security Incident Management**: Organizational Entities MUST ensure that logs are available and reliable to support incident management processes.

Wallet Provider Log Retention Policy
------------------------------------

Information related to the registration and management of Wallet Instances can be retained for up to 12 months after the deactivation/revocation of the Wallet or the associated User account, unless legal obligations require longer retention.

Credential Issuer Log Retention Policy
--------------------------------------

Credential Issuers define the retention period for Credentials based on sector-specific regulations. In the absence of specific regulations, the retention period for Credentials SHOULD not exceed 12 months after the date of expiration, configured at time of issuance within the metadata of the Issuer Signed part of the Credential.

Relying Party Log Retention Policy
----------------------------------

> Relying Parties MAY retain logs related to the processing of data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.

Reviewer 2: ```suggestion
Relying Parties MAY retain logs related to the processing of data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service or the expiration date of the presented Credentials, unless legal obligations require otherwise.
```

- Relying Parties SHOULD not log Credential presentation disclosure maps, where not necessary. 

Wallet Instances Logging Features
---------------------------------

> The Commission Implementing Regulation (EU) 2024/2979 establishes detailed rules for the application of Regulation (EU) No 910/2014, inclusing logging in the core functionalities of European Digital Identity Wallets. Please refer in particular to Recitals (11) and (13), and to Article 5e(4) and Article 6(2)(d), as well as Article 8(2)(d), which define logging obligations, user interface requirements, and data portability mechanisms.

Author: ```suggestion
In addition to the requirements included in the Consolidated Regulation (EU) No 910/2014, Art 5a 4(d) and the additional information provided within the Recitals (11) and (13), the Commission Implementing Regulation (EU) 2024/2979 establishes detailed rules for the application of Consolidated Regulation (EU) No 910/2014, inclusing logging in the core functionalities of European Digital Identity Wallets. 

Regulation (EU) 2024/2979, Article 9, which define logging obligations about Wallet Solutions and data portability mechanisms.
```

Wallet Provider Log Retention Policy
------------------------------------

Information related to the registration and management of Wallet Instances can be retained for up to 12 months after the deactivation/revocation of the Wallet or the associated User account, unless legal obligations require longer retention.

Credential Issuer Log Retention Policy
--------------------------------------

Credential Issuers define the retention period for Credentials based on sector-specific regulations. In the absence of specific regulations, the retention period for Credentials SHOULD not exceed 12 months after the date of expiration, configured at time of issuance within the metadata of the Issuer Signed part of the Credential.

Relying Party Log Retention Policy
----------------------------------

Relying Parties MAY retain logs related to the processing of data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.

> 

Author: ```suggestion
```

Relying Party Log Retention Policy
----------------------------------

Relying Parties MAY retain logs related to the processing of data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.

- Relying Parties SHOULD not log Credential presentation disclosure maps, where not necessary. 

Wallet Instances Logging Features
---------------------------------

The Commission Implementing Regulation (EU) 2024/2979 establishes detailed rules for the application of Regulation (EU) No 910/2014, inclusing logging in the core functionalities of European Digital Identity Wallets. Please refer in particular to Recitals (11) and (13), and to Article 5e(4) and Article 6(2)(d), as well as Article 8(2)(d), which define logging obligations, user interface requirements, and data portability mechanisms.

> ---------------------------------------
> 
> - ISO/IEC 27001:2024
> - Regulation (EU) No 910/2014 (eIDAS)
> - National regulations on traffic data retention.

Author: ```suggestion
```

Relying Party Log Retention Policy
----------------------------------

Relying Parties MAY retain logs related to the processing of data received from the Wallet only for the duration necessary to provide the requested service. The maximum retention period is 24 months after the conclusion of the service, the expiration, or the revocation date of the presented Credentials, unless legal obligations require otherwise.

- Relying Parties SHOULD not log Credential presentation disclosure maps, where not necessary. 

Wallet Instances Logging Features
---------------------------------

The Commission Implementing Regulation (EU) 2024/2979 establishes detailed rules for the application of Regulation (EU) No 910/2014, inclusing logging in the core functionalities of European Digital Identity Wallets. Please refer in particular to Recitals (11) and (13), and to Article 5e(4) and Article 6(2)(d), as well as Article 8(2)(d), which define logging obligations, user interface requirements, and data portability mechanisms.

> 

Author: ```suggestion
```
